server/core_self.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2017 The Nakama Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"go.uber.org/zap"
+	"strconv"
+	"strings"
+)
+
+type SelfUpdateOp struct {
+	UserId    []byte
+	Handle    string
+	Fullname  string
+	Timezone  string
+	Location  string
+	Lang      string
+	Metadata  []byte
+	AvatarUrl string
+}
+
+func SelfUpdate(logger *zap.Logger, db *sql.DB, updates []*SelfUpdateOp) (Error_Code, error) {
+	// Use same timestamp for all updates in this batch.
+	ts := nowMs()
+
+	// Start a transaction.
+	tx, e := db.Begin()
+	if e != nil {
+		logger.Error("Could not update user profile, transaction error", zap.Error(e))
+		return RUNTIME_EXCEPTION, errors.New("Could not update user profile")
+	}
+
+	var code Error_Code
+	var err error
+	defer func() {
+		if err != nil {
+			if rollbackErr := tx.Rollback(); rollbackErr != nil { // don't override value of err
+				logger.Error("Could not update user profile, rollback error", zap.Error(rollbackErr))
+			}
+		} else {
+			if e := tx.Commit(); e != nil {
+				logger.Error("Could not update user profile, commit error", zap.Error(e))
+				code = RUNTIME_EXCEPTION
+				err = errors.New("Could not update user profile, commit error")
```

Comment:
"Could not update user profile, commit error" -> "Could not update user profile"